### ISSUE TITLE
[fontconfig] build failure

### ISSUE THREAD

**Issue by jacobkahn** [microsoft/vcpkg#14864]
**Host Environment**
- OS: Ubuntu 18.04.5 LTS
- Compiler: gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0

**To Reproduce**
```
apt install -y automake autopoint gperf libtool # needed to build
```
install:
```
./vcpkg install fontconfig
```

This happens in a Google colab notebook environment in particular: https://colab.research.google.com/drive/16t70NhnnnxQhA3Nn5-6djFqGmkQatO7j?usp=sharing

**Failure logs** 
```
-- Performing post-build validation
There should be no empty directories in /content/vcpkg/packages/fontconfig_x64-linux
The following empty directories were found:

    /content/vcpkg/packages/fontconfig_x64-linux/share/fontconfig/locale/zh_CN/LC_MESSAGES

If a directory should be populated but is not, this might indicate an error in the portfile.
If the directories are not needed and their creation cannot be disabled, use something like this in the portfile to remove them:

    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/a/dir" "${CURRENT_PACKAGES_DIR}/some/other/dir")


Found 1 error(s). Please correct the portfile:
    /content/vcpkg/ports/fontconfig/portfile.cmake
-- Performing post-build validation done
Error: Building package fontconfig:x64-linux failed with: POST_BUILD_CHECKS_FAILED
```

**Additional context**
Looks like files in that directory are provided in the RPM package (https://centos.pkgs.org/8/centos-baseos-x86_64/fontconfig-2.13.1-3.el8.i686.rpm.html) [and other managers], so this seems like a port bug.

**Comment by Neumann-A** [comment]
I have the feeling that this happens due to gettext (msgfmt or similar) not being found on configure.

**Comment by JackBoosY** [comment]
On my machine:
```
autoreconf: Entering directory `.'
autoreconf: running: autopoint --force
autopoint: *** The AM_GNU_GETTEXT_VERSION declaration in your configure.ac
               file requires the infrastructure from gettext-0.19.8 but this version
               is older. Please upgrade to gettext-0.19.8 or newer.
autopoint: *** Stop.
autoreconf: autopoint failed with exit status: 1
```
We really need to add tool port `auto-tools` to vcpkg.

**Comment by Neumann-A** [comment]
change
https://github.com/microsoft/vcpkg/blob/65d742a57d9f6db031f8e9a356feb1dca4211cd3/ports/gettext/portfile.cmake#L59
to
`vcpkg_install_make() `
and it will install autopoint and the tools for localization. You then just need to put them on the path for the port.

**Comment by jacobkahn** [comment]
@JackBoosY — I reproduced the issue you saw with `AM_GNU_GETTEXT_VERSION` when trying to install the port on a clean Ubuntu 16.04 system. Either way, it seems these two issues are different but might have the same underlying cause. Can I help in any way? Is the `gettext` version issue going to be fixed?

**Comment by JackBoosY** [comment]
@jacobkahn According to the error message, we want to solve this issue from the surface is very simple: remove this empty folder.
But looking deeper, I want to confirm whether this will affect other parts of the configuration.

**Comment by JackBoosY** [comment]
In _SOURCE/po-config/Makefile.in.in_ line 298 and _SOURCE/po/Makefile.in.in_ line 297:
```
	      ln -s ../LC_MESSAGES/$(DOMAIN).mo $(DESTDIR)$(localedir)/$$lang/$$lc/$(DOMAIN).mo 2>/dev/null || \
	      ln $(DESTDIR)$(localedir)/$$lang/LC_MESSAGES/$(DOMAIN).mo $(DESTDIR)$(localedir)/$$lang/$$lc/$(DOMAIN).mo 2>/dev/null || \
	      cp -p $(DESTDIR)$(localedir)/$$lang/LC_MESSAGES/$(DOMAIN).mo $(DESTDIR)$(localedir)/$$lang/$$lc/$(DOMAIN).mo; \
	      echo "installing $$realcat link as $(DESTDIR)$(localedir)/$$lang/$$lc/$(DOMAIN).mo"; \
```

I think that's why this empty folder was installed.

**Comment by Neumann-A** [comment]
> I think that's why this empty folder was installed.

the folder is not empty for me but contains the zh_CN translation

**Comment by Neumann-A** [comment]
`installing ./zh_CN.gmo as /cygdrive/E/vcpkg_folders/fontconfig/packages/fontconfig_x64-windows/E/vcpkg_folders/fontconfig/installed/x64-windows/share/fontconfig/locale/zh_CN/LC_MESSAGES/fontconfig-conf.mo`

**Comment by JackBoosY** [comment]
@Neumann-A  The directory is also not empty on my machine.
@jacobkahn Can you find `fontconfig-conf.mo` in _VCPKG_ROOT/buildtrees/fontconfig/x64-linux-dbg_ ?

**Comment by jacobkahn** [comment]
@JackBoosY `fontconfig-conf.mo` isn't present — https://colab.research.google.com/drive/16t70NhnnnxQhA3Nn5-6djFqGmkQatO7j?usp=sharing.

Also, @JackBoosY @Neumann-A — is this other issue with an outdated version of gettext [that you mentioned here](https://github.com/microsoft/vcpkg/issues/14864#issuecomment-736317218) also going to be looked at? I actually am running into that issue as well in a different environment (Ubuntu 16.04). Should I open a separate issue for that?

**Comment by JackBoosY** [comment]
@jacobkahn Same problems should be record in one issue : )

**Comment by tlikhomanenko** [comment]
I have got the same error as @jacobkahn pointed. Repro steps for me in the docker image:

```
# start container with nvidia/cuda:10.1-cudnn7-devel-ubuntu18.04 and inside
apt-get update
APT_INSTALL="apt-get install -y --no-install-recommends" 

DEBIAN_FRONTEND=noninteractive $APT_INSTALL zlib1g-dev libcurl4-openssl-dev wget
# install cmake 3.10.3
cd /tmp && wget https://cmake.org/files/v3.10/cmake-3.10.3.tar.gz  &&     tar -xzvf cmake-3.10.3.tar.gz  && cd cmake-3.10.3  &&     ./bootstrap --system-curl &&     make -j$(nproc) &&  make install && cmake --version

$APT_INSTALL curl zip unzip tar g++ git pkg-config gperf autoconf autopoint libtool

cd /tmp && git clone https://github.com/microsoft/vcpkg
./vcpkg/bootstrap-vcpkg.sh
./vcpkg install fontconfig
```

**Comment by JackBoosY** [comment]
I don't know if this data file is important, if it's not, I can give you a patch to fix it.

**Comment by tlikhomanenko** [comment]
@JackBoosY I can try your patch to build it and then downstream packages I need and test if everything works if this will be helpful for you (at least unblock me on my own stuff).

**Comment by JackBoosY** [comment]
@tlikhomanenko  Please unzip [this file](https://github.com/microsoft/vcpkg/files/5657712/fontconfig.zip) to _VCPKG_ROOT/ports/fontconfig_ and try again.

**Comment by jacobkahn** [comment]
@JackBoosY — assuming the data file isn't important for most use cases, is it reasonable to expect a fix to the port soon, or should I remove the port from my dependency tree? This is affecting downstream users of my project's port who are reporting this failure.

**Comment by JackBoosY** [comment]
@jacobkahn Since the root cause has not been found, I am not sure if this is the best solution.

**Comment by vespakoen** [comment]
Looking at this: https://gitlab.freedesktop.org/fontconfig/fontconfig/-/blob/master/po-conf/zh_CN.po (Being the only translation present)
It seems to be translations for settings in `conf.d` files.
I don't think it's very helpful for libraries and I feel it can be safely removed.
